### ISSUE TITLE
Actions Logging

### ISSUE THREAD

**Issue by LukasRH** [RhythmGamesDev/RhythmGamesBot#13]
Log actions taken on the server. 

Events:
- message delete
- message edit
- member left
- member joined
- member updates
- channel updates
- channel creation
- channel delete
- possibly more

1. have command to select which channel to send logs in
2. have command to select which events to log

**Comment by HTGAzureX1212** [comment]
Not sure if we can have a configuration file instead of commands for storing which channels to log which events, as well as which events to log.

**Comment by LukasRH** [comment]
The events shouldn't be a command themself. The list is just to show what should be logged. 
And it will need some sort of storage to be persistent, yes. But I would like one single command to configure which events to log so the bot don't have to be changed each time we want to change that.

**Comment by HTGAzureX1212** [comment]
As from my past experiences of advanced moderation bots - taking HepBoat as an example - has a YAML configuration file to configure and audit the functionality of the bot. I mean technically we could have a file to load into memory when the bot starts for figuring out the logging functionality (such that we don't have to open, read the file and manipulate it just to get whether to log the event or where to log it).

**Comment by LukasRH** [comment]
it's fine to have a file that loads into memory at startup. but what if we don't want it to log members leaving while it's running. 
There need to be run-time configuration, is all I'm saying

**Comment by HTGAzureX1212** [comment]
We could perhaps implement a reloader...? Detect changes and reload the configuration? 🤔

**Comment by cgytrus** [comment]
A file watcher?

**Comment by LukasRH** [comment]
So to make changes, one should log in to the server and edit the configuration file?

**Comment by HTGAzureX1212** [comment]
Yeah, perhaps something like that. (Referring to cgytrus's comment)

**Comment by LukasRH** [comment]
yeah no. It should be possible through discord.

**Comment by cgytrus** [comment]
I agree with Lukas, runtime configuration and save it either when changes happen or when a command is execute (or both)

**Comment by HTGAzureX1212** [comment]
I updated my comment above 😂😂

**Comment by LukasRH** [comment]
Its not that hard a thing to make, I've done it before. I can take this task on my shoulders if it is

**Comment by HTGAzureX1212** [comment]
Hmm.. that's true but I am guessing we can make some sort of a reloadconfig command? 🤔

**Comment by LukasRH** [comment]
We won't need to reload the config if done as I suggest.

**Comment by HTGAzureX1212** [comment]
Got it! Thanks 👌

**Comment by The1Moxie** [comment]
> Its not that hard a thing to make, I've done it before. I can take this task on my shoulders if it is

I can help if needed, got bored of the TTT game you wanted me to make and been reading up on this and the documentation about events. and been trying it out on my own. Think I got the hang of events.

and commands to enable, disable these events and assign channel on where to post the information. The bool and channel id could be saved to a database or something else to make sure than the action log configurations are not lost if the bot needs to be restarted or crashed.
